spec/requests/transformation_mappings_spec.rb

```diff
@@ -0,0 +1,114 @@
+
+        get(api_transformation_mapping_url(nil, transformation_mapping))
+
+        expect(response.parsed_body).to include('id' => transformation_mapping.id.to_s)
+        expect(response).to have_http_status(:ok)
+      end
+    end
+
+    context "without an appropriate role" do
+      it "is forbidden" do
+        api_basic_authorize
+
+        get(api_transformation_mapping_url(nil, transformation_mapping))
+
+        expect(response).to have_http_status(:forbidden)
+      end
+    end
+  end
+
+  describe "POST /api/transformation_mappings" do
+    let(:cluster) { FactoryGirl.create(:ems_cluster) }
+    let(:cluster2) { FactoryGirl.create(:ems_cluster) }
+
+    context "with an appropriate role" do
+      it "can create a new transformation mapping" do
+        api_basic_authorize(collection_action_identifier(:transformation_mappings, :create))
+
+        new_mapping = {"name"                         => "new transformation mapping",
+                       "transformation_mapping_items" => [
+                         { "source" => api_cluster_url(nil, cluster), "destination" => api_cluster_url(nil, cluster2) }
+                       ]}
+        post(api_transformation_mappings_url, :params => new_mapping)
+
+        expected = {
+          "results" => [a_hash_including("href" => a_string_including(api_transformation_mappings_url),
+                                         "name" => "new transformation mapping")]
+        }
+        expect(response.parsed_body).to include(expected)
+        expect(response).to have_http_status(:ok)
+      end
+
+      it "will raise a bad request if mappings are not specified" do
+        api_basic_authorize(collection_action_identifier(:transformation_mappings, :create))
+
+        new_mapping = {"name" => "new transformation mapping"}
+        post(api_transformation_mappings_url, :params => new_mapping)
+
+        expected = {
+          "error" => a_hash_including(
+            "kind"    => "bad_request",
+            "message" => /Must specify transformation_mapping_items/
+          )
+        }
+        expect(response.parsed_body).to include(expected)
+        expect(response).to have_http_status(:bad_request)
+      end
+
+      it "will raise a bad request if source and destination are not specified for mappings" do
+        api_basic_authorize(collection_action_identifier(:transformation_mappings, :create))
+
+        new_mapping = {"name" => "new transformation mapping", "transformation_mapping_items" => [{}]}
+        post(api_transformation_mappings_url, :params => new_mapping)
+
+        expected = {
+          "error" => a_hash_including(
+            "kind"    => "bad_request",
+            "message" => /Must specify source and destination hrefs/
+          )
+        }
+        expect(response.parsed_body).to include(expected)
+        expect(response).to have_http_status(:bad_request)
+      end
+    end
```

Comment:
can you add a test for when the source or destination href is bogus, i.e. /api/invalid_collection/:id,  we should get a BadRequest on those.  Might need a minor change to fetch_mapping_resource. Thanks.